### PR TITLE
Credential status

### PR DESCRIPTION
**Pull request by osinakayah**

### CODE REVIEW

File: src/features/credentials/CredentialsScreen.js [hunk 1]
```diff
@@ -116,6 +116,28 @@ export const CredentialListItem = withErrorBoundary(
         </Stack>
 
         <NBox mt={4} flexDirection="row" alignItems={'flex-end'}>
+          {!isCredentialValid(credential) ? (
+            <NBox
```
Comment (maycon-mello) [inline src/features/credentials/CredentialsScreen.js:120]: Can we avoid the duplicated NBox and styling? perhaps something like this:
```
<NBox
    mr={1}
    style={{
      width: 12,
      height: 12,
      borderRadius: 6,
      backgroundColor: isCredentialValid(credential) ? Theme.colors.circleChecke : Theme.colors.error
    }}
  />
`

**Review by maycon-mello — CHANGES_REQUESTED** [review]
question about duplicated code

File: src/features/credentials/components/CredentialStatus.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import configureMockStore from 'redux-mock-store';
+import {CredentialStatus} from './CredentialStatus';
+import {renderAppProviders} from '../../../core/test-utils';
+
+const mockStore = configureMockStore();
+
+describe('CredentialsScreen', () => {
+  it('should render correctly', () => {
```
Comment (mike-parkhill) [inline src/features/credentials/components/CredentialStatus.test.js:10]: should probably have a snapshot for each of Verified, Invalid and Expired (and potentially Revoked if we can identify it)

File: src/features/credentials/components/CredentialStatus.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  const [status, setStatus] = useState(CREDENTIAL_STATUS.VERIFIED);
+
+  useEffect(() => {
+    isCredentialValid(credential).then(response => {
+      setStatus(response.status);
+    });
+  }, [credential]);
+
+  const statusData = useMemo(() => {
+    switch (status) {
```
Comment (mike-parkhill) [inline src/features/credentials/components/CredentialStatus.js:23]: are we able to differentiate Revoked from Invalid?

File: src/features/credentials/credentials.js [hunk 4]
```diff
@@ -246,3 +250,20 @@ export function isInThePast(date) {
   today.setHours(0, 0, 0, 0);
   return date < today;
 }
+
+export async function isCredentialValid(credential) {
+  const hasExpired = credential.expirationDate
+    ? isInThePast(new Date(credential.expirationDate))
```
Comment (mike-parkhill) [inline src/features/credentials/credentials.js:256]: what happens if `credential.expirationDate` doesn't represent a valid date/time? e.g. it's just text or some other object structure?

File: src/assets/icons/index.js [hunk 5]
```diff
@@ -32,7 +32,9 @@ import ChevronUpIcon from './chevron-up-icon.svg';
 import VerifiedIcon from './verified.svg';
 import InvalidIcon from './invalid.svg';
 import ExpiredIcon from './expired.svg';
+import RevokeIcon from './revoke.svg';
```
Comment (mike-parkhill) [inline src/assets/icons/index.js:35]: nitpick: should probably be `RevokedIcon`

File: src/features/credentials/hooks/credentialHooks.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        return {
+          message: translate('credentials.valid'),
+          icon: <VerifiedIcon />,
+          color: Theme.colors.circleChecked,
+        };
+    }
+  }, [status]);
+
+  useEffect(() => {
+    console.log(credential, 'credential');
```
Comment (mike-parkhill) [inline src/features/credentials/hooks/credentialHooks.js:45]: is this log still needed?

File: src/features/credentials/hooks/credentialHooks.test.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+describe('Credential  hooks', () => {
+  it('get credential status', async () => {});
```
Comment (mike-parkhill) [inline src/features/credentials/hooks/credentialHooks.test.js:2]: were you intending to add something here?

File: src/features/credentials/hooks/credentialHooks.js (hunk 6 above)
Comment (osinakayah) [reply src/features/credentials/hooks/credentialHooks.js:45]: No, it I will remove it in the next commit

File: src/features/credentials/hooks/credentialHooks.test.js (hunk 7 above)
Comment (osinakayah) [reply src/features/credentials/hooks/credentialHooks.test.js:2]: Yes, I am working on the unit test now.

**Review by mike-parkhill — APPROVED** [review]

**Review by maycon-mello — APPROVED** [review]

**Pull request merged**